control-plane/subcommand/server-acl-init/command.go

```diff
@@ -663,6 +668,35 @@ func (c *Command) Run(args []string) int {
 		}
 	}
 
+	if c.flagTelemetryCollector {
+		// the telemetry collector doesn't currently need any special permissions and relies on the rest of the
+		// control plane to interact directly with Consul (e.g. service registration).
+		// We're creating our own RBAC here  so that we can login using the component auth method instead of the
+		// client application one
```

Comment:
Hmmm If it's on the mesh, it needs `service:write` which anonymous token doesn't have.

Reply:
We're switching back to the generic service auth-method